Documentation/driver-api/ancillary_bus.rst

```diff
@@ -0,0 +1,200 @@
+mics/speakers etc. The split for the core's functionality can be arbitrary or
+be defined by the DSP firmware topology and include hooks for test/debug. This
+allows for the audio core device to be minimal and tightly coupled with
+handling the hardware-specific logic and communication.
+
+The ancillary bus is intended to be minimal, generic and avoid domain-specific
+assumptions. Each ancillary_device represents a part of its parent
+functionality. The generic behavior can be extended and specialized as needed
+by encapsulating an ancillary_device within other domain-specific structures and
+the use of .ops callbacks. Devices on the same ancillary bus do not share any
+structures and the use of a communication channel with the parent is
+domain-specific.
+
+When Should the Ancillary Bus Be Used
+=====================================
+
+When a kernel driver registers either an ancillary_device or ancillary_driver on
+the ancillary bus, we can use the nomenclature to refer to this kernel driver as
+a registering driver.  It is the entity that will allocate memoy for and
+register something on the ancillary bus.
+
+The ancillary bus is to be used when multiple registering drivers, who share a
+common header file, need a mechanism to make connection and provide access to a
+shared struct allocated by the registering driver(s) that registered the
+ancillary_device(s).  Each of these registering drivers can be from the same
+subsystem, or from multiple subsystems.
+
+The emphasis here is on a common generic interface that keeps subsystem
+customization out of the bus infrastructure.
+
+When Should the Ancillary Bus Not Be Used
+=========================================
+
+If there is a need for a heavier weight bus that provides functionality beyond
+just the matching up of multiple registering drivers, and sacrifices common
+generality to extend the bus functionality into subsystem specific functions,
+then the ancillary bus should not be used.  A custom bus solution, or some other
+method should be considered in this use case.
+
+Ancillary Devices
+=================
+
+A ancillary_device is created and registered to represent a part of its parent
+device's functionality. It is given a match_name that is used for driver
+binding and a release callback that is invoked when the device is unregistered.
+
+.. code-block:: c
+
+	struct ancillary_device {
+		struct device dev;
+		const char *match_name;
+		void (*release)(struct ancillary_device *);
+		u32 id;
+	};
+
+The ancillary device is enumerated when it is attached to the bus. The device
+is assigned a unique ID automatically that will be appended to its name. If
+two ancillary_devices both named "foo" are registered onto the bus, they will
+have the device names, "foo.x" and "foo.y", where x and y are unique integers.
+
+Ancillary Device Memory Model and Lifespan
+------------------------------------------
+
+There will be a parent struct, defined in the shared header file, that contains
```

Comment:
avoid 'there will be', use active form:

A parent struct, defined... contains the ancillary_device

Reply:
again I like parent object, shared object better